src/main/kotlin/com/exactpro/th2/codec/fixng/FixNgCodec.kt

```diff
@@ -228,20 +277,35 @@ class FixNgCodec(dictionary: IDictionaryStructure, settings: FixNgCodecSettings)
 
     private val preparedHeaderTags = arrayOf(TAG_BEGIN_STRING, TAG_BODY_LENGTH, TAG_MSG_TYPE)
 
-    private fun Message.decode(source: ByteBuf, bodyDef: Message, isDirty: Boolean, dictionaryFields: Map<Int, Field>, context: IReportingContext): MutableMap<String, Any> = mutableMapOf<String, Any>().also { map ->
+    private fun Message.decode(
+        source: ByteBuf,
+        bodyDef: Message,
+        isDirty: Boolean,
+        dictionaryFields: Map<Int, Field>,
+        context: IReportingContext
+    ): MutableMap<String, Any> = mutableMapOf<String, Any>().also { map ->
         val tagsSet: MutableSet<Int> = hashSetOf(*preparedHeaderTags)
         val usedComponents = mutableSetOf<String>()
+        val isNextPartTag: (tag: Int) -> Boolean = { tag ->
+            when (this) {
+                headerDef -> bodyDef[tag] ?: trailerDef[tag]
+                trailerDef -> null
+                else -> trailerDef[tag]
+            } != null
+        }
 
         source.forEachField(
             decodeDelimiter, charset, isDirty,
-            { handleError(isDirty, context, it) }
+            { tag, warn ->
+                if (isNextPartTag(tag)) {
+                    LOGGER.warn { "Tag ($tag) reading problem: $warn" }
+                } else {
+                    handleError(isDirty, context, warn)
+                }
```

Comment:
question: why do we have a condition for when to report the warning as an event and when not to? I try to understand what it gives us and can't... Could you please add a bit of context here?

Reply:
I try to split error / warning for different message parts. 
For example, codec process body part of message and found tag '010' from trailer. In this case process method is terminated because 10 tag is part of trailer. Codec can print warning into log because it will face to the same problem when it process '010' tag in trailer part and call handleError for that case